### PR TITLE
codeforpoznan.pl_v2 static frontend infrastructure

### PR DESCRIPTION
**Pull request by magul**
First draft for infrastructure needed to serve static assets for `codeforpoznan.pl_v2`.

It's still lacking permissions to update Cloudformation for `codeforpoznan.pl_v2` user. Eventually, plan is to run something like that on Travis:
```
aws s3 cp ...
aws cloudfront update-distribution ...
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+provider "aws" {
+  region  = "eu-west-1"
+  profile = "codeforpoznan"
+}
+
+provider "aws" {
+  alias   = "north_virginia"
+  region  = "us-east-1"
```
Comment (arturtamborski) [inline main.tf:17]: why US?

**Review by arturtamborski — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (magul) [reply main.tf:17]: > If you want to require HTTPS between viewers and CloudFront, you must change the AWS region to US East (N. Virginia) in the AWS Certificate Manager console before you request or import a certificate.

see: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cnames-and-https-requirements.html#https-requirements-aws-region

Comment (magul) [reply main.tf:17]: + be aware that there will be additional commit that will add CloudFront update permission to `codeforpoznanpl_v2` user (as we will attach it credentials to Travis to do routine updates).

**Review by Maciek246 — APPROVED** [review]

**Comment by magul** [comment]
@arturtamborski @Maciek246 

I have to change the way we treat static assets. My initial approach was to put every version in separate directories and then change the origin path in CloudFront distribution. That could work if we would treat Terraform script only as a bootstrap (meaning we will not use that in future), but if we're dedicated to using it, we will change something in CloudFront distribution using boto and that will create configuration drift (meaning that actual state of our infrastructure will be different then `tfstate` and `tf` files).

Another approach that would work here would be to allow `codeforpoznan_pl_v2` user to run only specific parts of Terraform infrastructure (that will also change `tfstate` used by main Terraform user), but I cannot find any pieces of information about such approach.

Right now (as I already run this Terraform script) we need to add to codeforpoznan.pl_v2 repository's Travis script these two lines:
```
aws --profile codeforpoznan_pl_v2 s3 sync . s3://codeforpoznan-public/codeforpoznan.pl_v2/ --delete
aws --profile codeforpoznan_pl_v2 cloudfront create-invalidation --distribution-id ER7F0YI4881NW --paths "/*"

```

**Review by arturtamborski — APPROVED** [review]

**Pull request merged**